packages/calypso-e2e/src/lib/pages/published/published-posts-list-page.ts

```diff
@@ -24,6 +24,15 @@ export class PublishedPostsListPage extends BaseContainer {
 		super( page, selectors.page );
 	}
 
+	/**
+	 * Post-object creation initialization steps.
+	 *
+	 * @returns {Promise<void>} No return value.
+	 */
+	async _postInit(): Promise< void > {
+		await this.page.waitForSelector( selectors.posts );
```

Comment:
I can see this pattern repeating, where we wait for additional elements in the `_postInit` hook. Do you think making the `BaseContainer` accept an array of selectors that need to be awaited before initialization would help here? Its `_init` method could then be:

```js
	async _init(): Promise< void > {
		await this._visit();
		await this._preInit();
		await this.page.waitForLoadState( 'load' );
		if ( this.selectors.length > 0 ) {
			await Promise.all(
				this.selectors.map( selector => this.page.waitForSelector( selector ) )
			);
		}
		await this._postInit();
	}
```

Reply:
You are right, I do frequently use the `_postInit` more or less to wait on additional elements. 

I believe the Selenium counterpart accepts a list of elements. I'll spin up a new task on the board to alter this behavior. How does that aound?

Reply:
Filed #53557 for the task.